plugins/org.jboss.tools.fuse.reddeer/src/org/jboss/tools/fuse/reddeer/editor/CamelEndpointDialog.java

```diff
@@ -0,0 +1,151 @@
+package org.jboss.tools.fuse.reddeer.editor;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.jboss.reddeer.common.wait.WaitUntil;
+import org.jboss.reddeer.common.wait.WaitWhile;
+import org.jboss.reddeer.core.condition.ShellWithTextIsAvailable;
+import org.jboss.reddeer.swt.api.Button;
+import org.jboss.reddeer.swt.api.TreeItem;
+import org.jboss.reddeer.swt.impl.button.CheckBox;
+import org.jboss.reddeer.swt.impl.button.PushButton;
+import org.jboss.reddeer.swt.impl.group.DefaultGroup;
+import org.jboss.reddeer.swt.impl.shell.DefaultShell;
+import org.jboss.reddeer.swt.impl.text.LabeledText;
+import org.jboss.reddeer.swt.impl.tree.DefaultTree;
+import org.jboss.reddeer.swt.impl.tree.DefaultTreeItem;
+
+/**
+ * Manipulates with dialog (Wizard) for adding Global Camel Endpoints
+ * 
+ * @author djelinek
+ */
+public class CamelEndpointDialog {
```

Comment:
Please use a parent class WizardDialod. Then, you needn't to implement methods such as cancel() or finish().